### ISSUE TITLE
[BETA][EAS] Filters are Misleading/Undocumented

### ISSUE THREAD

**Issue by sirajrauff** [chef/automate#2622]
Making use of `/v1/applications` as part of the Beta Automate API for EAS and encountered a few bumps trying to get it working.

First and foremost is there's no documentation regarding the endpoints other than in [V1 Policies](https://automate.chef.io/docs/iam-v1-policies/#applications-beta) (there's nothing for V2). This provides the permissions required but nowhere else can pagination or filtering arguments be found, which led to me initially think the endpoint was bugged (#2610). I ended up using the source code [[1](https://github.com/chef/automate/blob/master/components/automate-chef-io/data/docs/api_chef_automate/applications/applications.swagger.json#L70)],[[2](https://github.com/chef/automate/blob/7d821f9ddf957ef9dee7413874710e9a323fe433/components/applications-service/pkg/storage/postgres/services.go#L424)] to get my code working.

Second, the filters seem to correspond more to Compliance than to Habitat Services. Consider this return object:
```
{
  "services": [
    {
      "supervisor_id": "0710a75e58fc460ca5861559deac1d14",
      "release": "sirajrauff-public/sample-node-app/0.1.4/20191218214052",
      "group": "sample-node-app.default",
      "health_check": "OK",
      "status": "RUNNING",
      "application": "sample-node-app",
      "environment": "local-test",
      "fqdn": "2531d46609d9",
      "channel": "stable",
      "update_strategy": "AT-ONCE",
      "site": "laptop",
      "previous_health_check": "NONE",
      "current_health_since": "15 hours 16 minutes",
      "health_updated_at": "2020-01-14T01:05:58.885441Z",
      "disconnected": true,
      "last_event_occurred_at": "2020-01-14T01:07:28.965647Z",
      "last_event_since": "15 hours 14 minutes"
    },
    ...
  ]
}
```
- The `status` filter corresponds to [health check](https://github.com/chef/automate/blob/7d821f9ddf957ef9dee7413874710e9a323fe433/components/applications-service/pkg/storage/postgres/services.go#L428-L429), which is misleading as the object has both `status` and `health_check`. i.e. the first time I saw the `status` filter, I tried to do `status:RUNNING` which is incorrect.
- The [`buildstamp`](https://github.com/chef/automate/blob/7d821f9ddf957ef9dee7413874710e9a323fe433/components/applications-service/pkg/storage/postgres/services.go#L443-L444) filter corresponds to [Release](https://www.habitat.sh/docs/glossary/#glossary-artifacts) and should be called `release`. In the same vein, [`service`](https://github.com/chef/automate/blob/7d821f9ddf957ef9dee7413874710e9a323fe433/components/applications-service/pkg/storage/postgres/services.go#L455-L456) corresponds to [name](https://www.habitat.sh/docs/glossary/#glossary-artifacts). I would've assumed service is `sample-node-app.default`, i.e `<name>.<group>`, not just the name
- `release` is the [full identifier](https://www.habitat.sh/docs/glossary/#glossary-artifacts). Would've been nice if this was renamed `ident` or `identifier` and an additional property `release` was returned with only the buildstamp/release.
- `group` searches by the [suffix](https://github.com/chef/automate/blob/7d821f9ddf957ef9dee7413874710e9a323fe433/components/applications-service/pkg/storage/postgres/services.go#L452-L453), i.e. sample-node-app.**default**, but `group` in the object returned is in the format `<name>.<group>`. This would be more clear if only the suffix was returned in the property or if both were used in the filtering. This also means you can't use group as it exists in the return object and use it in subsequent filters, you'd have to do a `split('.')`

Misc:
- Would be nice to have `disconnected` be a filter, I often don't care about disconnected services and want to see what version/releases of a particular service are currently running

**Comment by danielsdeleo** [comment]
Hi, thanks for filing this.

To your first point, I'll file an internal issue to make the API docs more complete. I as mentioned elsewhere, we are working to publish API docs online using the swagger stuff as the basis.

For the other stuff:
* `status`: for filtering this is a combined concept covering both health checks and disconnectedness currently. Using `status:disconnected`returns only the disconnected services.
* identifier/buildstamp/group/etc.: Some of this naming predates the habitat project standardizing some of the terms. We selected the filter-able fields based on what we thought would simplify the most common use-cases and match the labels in the UI as best as possible. It's unlikely we're going to prioritize changing these as it's a pretty high churn but low visibility task.
* `disconnected` filter: As I mentioned, we have a filter to select the services that are disconnected. Sounds like you're asking to exclude them. The way the status filter currently works doesn't support that. We have explored some other ways of structuring this to give you more granular control, but haven't committed to that work at this time. We do plan to reintroduce the ability to manually remove disconnected services without waiting for the automatic deletion to clean them up.
